src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1642,6 +1648,7 @@ private void ProcessReaderThread(object state)
                     else
                     {
                         // Normal output data.
+                        if (!_connectionEstablished) { _connectionEstablished = true; }
```

Comment:
should this just be simply `_connectionEstablished = true`

Reply:
I prefer to do the assignment only once in the loop.